source/extensions/filters/udp/dns_filter/dns_parser.h

```diff
@@ -74,26 +76,42 @@ class DnsAnswerRecord : public BaseDnsRecord {
 using DnsAnswerRecordPtr = std::unique_ptr<DnsAnswerRecord>;
 using DnsAnswerMap = std::unordered_multimap<std::string, DnsAnswerRecordPtr>;
 
+/**
+ * @brief This struct is used to hold pointers to the counters that are relevant to the
+ * parser. This is done to prevent dependency loops between the parser and filter headers
+ */
+struct DnsParserCounters {
+  Stats::Counter* underflow_counter;
+  Stats::Counter* record_name_overflow;
+  Stats::Counter* query_parsing_failure;
```

Comment:
nit: use references here to avoid being null

Reply:
Updated